source/client/HandView.py

```diff
@@ -85,300 +82,94 @@ def update(self):
         if len(self.current_hand) == 0:
             self.hand_info = []
         elif not self.last_hand == self.current_hand:
-            self.hand_info = self.wrapHand(self.current_hand, self.hand_info)
-        self.showHolding(self.hand_info)  # displays hand
+            self.hand_info = HandManagement.wrapHand(self, self.current_hand, self.hand_info)
+        HandManagement.showHolding(self, self.hand_info)  # displays hand
         if self.refresh_flag:  # if needed to rescale card size, then refreshXY again.
-            self.hand_info = self.refreshXY(self.hand_info)
-        # display draw pile and various action buttons
-        loc_xy = (self.draw_pile.x, self.draw_pile.y)
-        self.draw_pile.draw(self.display, loc_xy, self.draw_pile.outline_color)
-        # update discard info and redraw
-        discard_info = self.controller.getDiscardInfo()
-        self.top_discard = discard_info[0]
-        self.pickup_pile_sz = discard_info[1]
-        if self.pickup_pile_sz > 0:
-            self.top_discard_wrapped = UICardWrapper(self.top_discard, (100, 25), UIC.scale)
-            self.pickup_pile = self.top_discard_wrapped.img_clickable
-            loc_xy = (self.pickup_pile.x, self.pickup_pile.y)
-            self.pickup_pile.draw(self.display, loc_xy, self.pickup_pile.outline_color)
-            self.labelMedium(str(self.pickup_pile_sz), 150, 35)
-        if self.controller._state.round == -1 :
-            self.ready_yes_btn.draw(self.display, self.ready_yes_btn.outline_color)
-            self.ready_no_btn.draw(self.display, self.ready_no_btn.outline_color)
-        self.sort_status_btn.draw(self.display, self.sort_status_btn.outline_color)
-        self.sort_btn.draw(self.display, self.sort_btn.outline_color)
-        self.prepare_card_btn.draw(self.display, self.prepare_card_btn.outline_color)
-        self.clear_prepared_cards_btn.draw(self.display, self.clear_prepared_cards_btn.outline_color)
-        self.play_prepared_cards_btn.draw(self.display, self.play_prepared_cards_btn.outline_color)
-        self.discard_action_btn.draw(self.display, self.discard_action_btn.outline_color)
+            self.hand_info = HandManagement.refreshXY(self, self.hand_info)
+        HandAndFootButtons.ButtonDisplay(self)
 
     def nextEvent(self):
         """This submits the next user input to the controller,
 
         key strokes don't do anything unless designating values for prepared wild cards,
         at which time the mouse is ignored unless you want to clear the prepared cards."""
 
-        for event in pygame.event.get():
+        for self.event in pygame.event.get():
```

Comment:
event does not need and should not have this self prefix - it is not a member of the hand_view class

Reply:
This occurred when I moved helper methods to another file.  I have changed "self" to "hand_view" in the various helper methods in this file, as suggested.